src/app/services/data/origin-destination.service.ts

```diff
@@ -27,6 +28,18 @@ export class OriginDestinationService {
     private trainrunService: TrainrunService,
   ) {}
 
+  /**
+   * Returns the origin/destination nodes used for output.
+   *
+   * Note that other nodes may be used for the calculation.
```

Comment:
```suggestion
   * Returns the origin/destination nodes used for output.
   * Also handles the nodes selection (`getSelectedNodes()`) and nodes active filtering (`getVisibleNodes()`) when computing origin/destination algorithm.
   * Note that other nodes may be used for the calculation.
```

It could be useful to precise why `getSelectedNodes` and `getVisibleNodes` are used.